src/poetry/core/packages/utils/utils.py

```diff
@@ -133,19 +132,19 @@ def is_python_project(path: Path) -> bool:
     return pyproject or setuptools_project
 
 
-def is_archive_file(name: str) -> bool:
+def is_archive_file(name: Path) -> bool:
     """Return True if `name` is a considered as an archive file."""
     ext = splitext(name)[1].lower()
     if ext in ARCHIVE_EXTENSIONS:
         return True
     return False
 
 
-def splitext(path: str) -> tuple[str, str]:
+def splitext(path: Path) -> tuple[str, str]:
     """Like os.path.splitext, but take off .tar too"""
-    base, ext = posixpath.splitext(path)
+    base, ext = path.stem, path.suffix
```

Comment:
That's not the same:

```
>>> posixpath.splitext('foo/bar.baz')
('foo/bar', '.baz')
>>> path = Path('foo/bar.baz')
>>> path.stem, path.suffix
('bar', '.baz')
```

However, it does not matter for our uses so it might still be ok (if we fix the docstring).

On the other side, we could change the signature to allow `str | Path` and keep the old implementation. We don't even need a `noqa` as it seems. I slightly tend to this variant. Then, we don't have to convert the `str` to `Path` in `link.py`.